common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1519,33 +1519,35 @@ def create_child(self, user_id, parent_usage_key, block_type, block_id=None, fie
                 user_id, parent_usage_key.course_key, block_type, block_id=block_id, fields=fields,
                 **kwargs)
 
-            # don't version the structure as create_item handled that already.
-            new_structure = self._lookup_course(xblock.location.course_key).structure
+            # attach to parent if given
+            if 'detached' not in xblock._class_tags:    # pylint: disable=protected-access
```

Comment:
If you instead use the condition to short-circuit the method, you have to change the indentation less:

    if 'detached' in xblock._class_tags:
        return xblock
    ...